ktor-server/ktor-server-netty/jvm/src/io/ktor/server/netty/cio/NettyResponsePipeline.kt

```diff
@@ -6,138 +6,94 @@ package io.ktor.server.netty.cio
 
 import io.ktor.http.*
 import io.ktor.server.netty.*
-import io.ktor.server.netty.http2.*
 import io.ktor.util.*
 import io.ktor.util.cio.*
 import io.ktor.utils.io.*
-import io.netty.buffer.*
 import io.netty.channel.*
 import io.netty.handler.codec.http.*
 import io.netty.handler.codec.http2.*
 import kotlinx.coroutines.*
-import kotlinx.coroutines.CancellationException
-import kotlinx.coroutines.channels.*
 import java.io.*
-import java.util.*
+import java.util.concurrent.atomic.*
 import kotlin.coroutines.*
 
 private const val UNFLUSHED_LIMIT = 65536
 
+/**
+ * Contains methods for handling http request with Netty
+ * @param context
+ * @param coroutineContext
+ * @param activeRequests
+ * @param isCurrentRequestFullyRead
+ * @param isChannelReadComplete
+ */
 @OptIn(InternalAPI::class)
 internal class NettyResponsePipeline constructor(
-    private val dst: ChannelHandlerContext,
-    initialEncapsulation: WriterEncapsulation,
-    private val requestQueue: NettyRequestQueue,
-    override val coroutineContext: CoroutineContext
+    private val context: ChannelHandlerContext,
+    override val coroutineContext: CoroutineContext,
+    private val activeRequests: AtomicLong,
+    private val isCurrentRequestFullyRead: AtomicBoolean,
+    private val isChannelReadComplete: AtomicBoolean
```

Comment:
Not sure what exactly does it mean. Maybe one of these?
`isChannelReadCompletely`
`isReadChannelCompleted`

Reply:
Renamed to `isChannelReadCompleted`